package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1253,11 +1262,50 @@ class Tempfactors(AtomAttr):
     singular = 'tempfactor'
     per_object = 'atom'
     dtype = float
+    transplants = defaultdict(list)
+    aliases = [{"attrname": "bfactors", "singular": "bfactor"}]
 
     @staticmethod
     def _gen_initial_values(na, nr, ns):
         return np.zeros(na)
 
+    def bfactor(self):
+        """Tempfactor alias property for atom
+
+        .. versionadded:: 2.0.0
+        """
+        return self.universe.atoms[self.ix].tempfactor
```

Comment:
I _think_ the discussion was to add a deprecation warning on access to bfactor/bfactors? Not sure how necessary that is.

Reply:
This can be easily implemented but again not sure if it's necessary. @MDAnalysis/coredevs ?

Reply:
Having `aliases` might invite other shenanigans like this one. I'd prefer to get rid of bfactors in 3.0.0, though, and rather have one way to access data.

Cheap appeal to [The Zen of Python](https://www.python.org/dev/peps/pep-0020/) _There should be one-- and preferably only one --obvious way to do it._

(Yeah, there's also "In the face of ambiguity, refuse the temptation to guess." ...)

Reply:
To that end I would:
* remove the `aliases` construct and just manually do that in the TopologyAttr metaclass, with a note to remove on 3
* Add a deprecation warning (as mentioned by @IAlibay) that `bfactors` is going away in 3.0.0 and is currently only a wrapper around Tempfactors